src/test/rustdoc/inline_cross/assoc-items.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// aux-build:assoc-items.rs
+// build-aux-docs
+// ignore-cross-compile
+
+#![crate_name = "foo"]
+
+extern crate assoc_items;
+
+// @has foo/struct.MyStruct.html
+// @!has - 'PrivateConst'
+// @has - '//*[@id="associatedconstant.PublicConst"]' 'pub const PublicConst: u8'
+// @has - '//*[@class="docblock"]' 'PublicConst: u8 = 123'
+// @has - '//*[@class="docblock"]' 'docs for PublicConst'
+// @!has - 'private_method'
```

Comment:
Does this form of `@!has` test work? I would assume that this form was looking for an XPath, not a text string.

Reply:
This form isn't looking for an XPath https://github.com/rust-lang/rust/blob/0a2e9ade83ff253bb489c63a95b1f499c5e0916f/src/etc/htmldocck.py#L51

Reply:
Huh, nice. I guess i'm not used to seeing that form.